pkg/operator/scheme.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operator
+
+import (
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+
+	securityv1 "github.com/openshift/api/security/v1"
+
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	nodeobservabilityv1alpha1 "github.com/openshift/node-observability-operator/api/v1alpha2"
+)
+
+var (
+	// scheme contains all the API types necessary for the operator's dynamic
+	// clients to work. Any new non-core types must be added here.
+	//
+	// NOTE: The discovery mechanism used by the client doesn't automatically
+	// refresh, so only add types here that are guaranteed to exist before the
+	// operator starts.
+	scheme = runtime.NewScheme()
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(nodeobservabilityv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(securityv1.AddToScheme(scheme))
+	utilruntime.Must(rbacv1.AddToScheme(scheme))
+	utilruntime.Must(mcv1.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
```

Comment:
I think it's better to be kept in `main.go` as kubebuilder scaffolding expects it there. In ExtDNS Operator it was put into a package to be able to be used via `GetOperatorScheme` function, there is no such a need for Nob Operator.

Reply:
Sorry, my bad, you need it in `operator` package to pass to the manager options.